src/Services/LDAPService.php

```diff
@@ -546,17 +548,31 @@ public function updateMemberFromLDAP(Member $member, $data = null, $updateGroups
         $member->LastSynced = (string)DBDatetime::now();
 
         foreach ($member->config()->ldap_field_mappings as $attribute => $field) {
+            // Special handling required for attributes that don't exist in the response.
             if (!isset($data[$attribute])) {
-                $this->getLogger()->notice(
-                    sprintf(
-                        'Attribute %s configured in Member.ldap_field_mappings, ' .
-                                'but no available attribute in AD data (GUID: %s, Member ID: %s)',
-                        $attribute,
-                        $data['objectguid'],
-                        $member->ID
-                    )
-                );
+                // a attribute we're expecting is missing from the LDAP response
+                if ($this->config()->get("reset_missing_attributes")) {
+                    // (Destructive) Reset the corresponding attribute on our side if instructed todo so.
+                    if (method_exists($member->$field, "delete") && $member->$field->exists()) {
+                        $member->$field->deleteFile();
```

Comment:
This only checks for `delete()` but then calls `deleteFile()`, meaning that it only actually works for the `thumbnailphoto` attribute (or any other `File` reference), right?

Reply:
Yeah, deleteFile is a method on DBFile which is the storage type of the object. In this case we delete the underlying file, then the reference. Unfortunately there's no `isComponent()` or an interface to depend on (that i know of). 

Suggestions welcome

Reply:
i've added a check for the other function.